src/Components/Web.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -25,9 +25,14 @@ export class BrowserRenderer {
     attachNavigationManagerToEventDelegator(this.eventDelegator);
   }
 
-  public attachRootComponentToLogicalElement(componentId: number, element: LogicalElement): void {
+  public attachRootComponentToLogicalElement(componentId: number, element: LogicalElement, appendContent: boolean): void {
     this.attachComponentToElement(componentId, element);
-    rootComponentsPendingFirstRender[componentId] = element;
+
+    // If we want to preserve existing HTML content of the root element, we don't apply the mechanism for
+    // clearing existing children. Rendered content will then append rather than replace the existing HTML content.
+    if (!appendContent) {
+      rootComponentsPendingFirstRender[componentId] = element;
```

Comment:
If we're doing this, can we rename the collection to something like `elementsToClearOnRootComponentRender`? Otherwise other code might assume that `rootComponentsPendingFirstRender` always contains all the root components that haven't rendered yet.